src/components/transactions/FilterModal.tsx

```diff
@@ -0,0 +1,318 @@
+        </CloseButtonContainer>
+
+        <CenteredContainer>
+          <H4CardNavTab>Filter Invoices</H4CardNavTab>
+        </CenteredContainer>
+
+        <SubheadingContainer>
+          <OneLine>
+            <Body1SemiboldText>Filter by date</Body1SemiboldText>
+            <ClearButton
+              isDisabled={
+                !(
+                  filterState.inProgressMinDateIsSet ||
+                  filterState.inProgressMaxDateIsSet
+                )
+              }
+              onPress={() => {
+                filterDispatch({
+                  type: 'CLEAR_DATE_FILTERS',
+                });
+              }}
+            />
+          </OneLine>
+        </SubheadingContainer>
+
+        <OneLine>
+          <LeftAlignContainer>
+            <FilterField
+              isSelected={filterState.inProgressMinDateIsSet}
+              onPress={() => setMinDatePickerIsVisible(true)}
+              width="93%"
+              minWidth={148}
+              useCalendarIcon
+            >
+              <Body1Text>
+                {filterState.inProgressMinDateIsSet ? (
+                  moment(filterState.inProgressMinDate).format('M/DD/YYYY')
+                ) : (
+                  <MidGrayText>From</MidGrayText>
+                )}
+              </Body1Text>
+            </FilterField>
+          </LeftAlignContainer>
+          <RightAlignContainer>
+            <FilterField
+              isSelected={filterState.inProgressMaxDateIsSet}
+              onPress={() => setMaxDatePickerIsVisible(true)}
+              width="93%"
+              minWidth={148}
+              useCalendarIcon
+            >
+              <Body1Text>
+                {filterState.inProgressMaxDateIsSet ? (
+                  moment(filterState.inProgressMaxDate).format('M/D/YY')
+                ) : (
+                  <MidGrayText>To</MidGrayText>
+                )}
+              </Body1Text>
+            </FilterField>
+          </RightAlignContainer>
+        </OneLine>
+        {minDatePickerIsVisible ? (
+          <RNDateTimePicker
+            value={filterState.inProgressMinDate}
+            onChange={e => {
+              if (e.type === 'set' && e.nativeEvent.timestamp) {
+                filterDispatch({
+                  type: 'SET_MIN_DATE',
+                  date: new Date(e.nativeEvent.timestamp),
+                });
+              }
+              setMinDatePickerIsVisible(false);
+            }}
+          />
+        ) : null}
+        {maxDatePickerIsVisible ? (
+          <RNDateTimePicker
+            value={filterState.inProgressMaxDate}
+            onChange={e => {
+              if (e.type === 'set' && e.nativeEvent.timestamp) {
+                filterDispatch({
+                  type: 'SET_MAX_DATE',
+                  date: new Date(e.nativeEvent.timestamp),
+                });
+              }
+              setMaxDatePickerIsVisible(false);
+            }}
+          />
+        ) : null}
```

Comment:
On iOS, separate date selection buttons appear when the filter by date "From" and "To" options are selected. You can then press these buttons and select dates after they show up. Would it be possible to trigger the data selection modals when our custom button is pressed?

Reply:
Also, the clear button doesn't seem to clear date selections.